### PR TITLE
Add Device Image to Entity Registry

### PR DESCRIPTION
**Pull request by bafonins**
#### Summary

Closes https://github.com/TheThingsNetwork/lorawan-stack/issues/1506

#### Changes

- Add support for including pictures for end devices on create/update operations
- Add separate bucket configuration for end device pictures
- Generate api
- CLI support for end device pictures

#### Notes for Reviewers

~* CLI supports is added to make testing easier, later I will create a separate PR once we are fine with the draft~
~* Will add docs once we are fine with the draft~

@mjamescompton please verify that this works as needed

You can test the feature by adding the `--picture` flag when creating/updating devices

#### Checklist

- [x] Scope: The referenced issue is addressed, there are no unrelated changes.
- [x] Compatibility: The changes are backwards compatible with existing API, database and configuration, according to the stability commitments in `README.md`.
- [x] Testing: The changes are covered with unit tests. The changes are tested manually as well.
- [x] Documentation: Relevant documentation is added or updated.
- [x] Changelog: Significant features, behavior changes, deprecations and fixes are added to `CHANGELOG.md`.
- [x] Commits: Commit messages follow guidelines in `CONTRIBUTING.md`, there are no fixup commits left.

### CODE REVIEW

**Comment by mjamescompton** [comment]
@bafonins I get this error when trying to create a new device in the console

```  ERROR Finished unary call duration=107.09838ms error=error:pkg/identityserver/store:database (database error) error_cause=pq: column "picture_id" does not exist error_correlation_id=d09091f4938f4efba1dc6485ad26e69d error_name=database error_namespace=pkg/identityserver/store grpc_code=Internal grpc_method=Create grpc_service=ttn.lorawan.v3.EndDeviceRegistry namespace=grpc request_id=01DT6SYA8Q2W2R5XX59W3QZ7DZ ```

File: pkg/identityserver/profile_picture.go [hunk 1]
```diff
@@ -32,10 +32,13 @@ import (
 )
 
 const maxProfilePictureStoredDimensions = 1024
+const maxEndDevicePictureStoredDimensions = 1024
 
 var profilePictureDimensions = []int{64, 128, 256, 512}
+var endDevicePictureDimensions = []int{64, 128, 256, 512}
 
 var profilePictureRand = rand.New(randutil.NewLockedSource(rand.NewSource(time.Now().UnixNano())))
+var endDevicePictureRand = rand.New(randutil.NewLockedSource(rand.NewSource(time.Now().UnixNano())))
```
Comment (htdvisser) [inline pkg/identityserver/profile_picture.go]: can use a single `pictureRand`

File: pkg/identityserver/profile_picture.go [hunk 2]
```diff
@@ -97,3 +100,50 @@ func (is *IdentityServer) processUserProfilePicture(ctx context.Context, usr *tt
 
 	return
 }
+
+func (is *IdentityServer) processEndDevicePicture(ctx context.Context, dev *ttnpb.EndDevice) (err error) {
```
Comment (htdvisser) [inline pkg/identityserver/profile_picture.go]: Let's rename this file from `profile_picture.go` to `picture.go`

**Comment by bafonins** [comment]
@htdvisser I added fix for https://github.com/TheThingsNetwork/lorawan-stack/pull/1624

File: cmd/ttn-lw-cli/commands/end_devices.go [hunk 3]
```diff
@@ -43,6 +43,7 @@ var (
 	selectEndDeviceFlags     = &pflag.FlagSet{}
 	setEndDeviceFlags        = &pflag.FlagSet{}
 	endDeviceFlattenPaths    = []string{"provisioning_data"}
+	pictureFlags             = &pflag.FlagSet{}
```
Comment (htdvisser) [inline cmd/ttn-lw-cli/commands/end_devices.go]: `pictureFlags` lives in the `commands` package, so if you only use it for end device pictures, you need to call it `endDevicePictureFlags`.

File: doc/content/reference/configuration/identity-server.md [hunk 4]
```diff
@@ -89,6 +89,12 @@ It is also possible to use [Gravatar](https://gravatar.com) for profile pictures
 
 - `is.profile-picture.use-gravatar`: Use Gravatar fallback for users without profile picture
 
+## End Device Picture Storage Options
+Similar to profile pictures, end devices can have pictures associated with them.
```
Comment (htdvisser) [inline doc/content/reference/configuration/identity-server.md]: Separate h1 and p with empty line

File: pkg/identityserver/picture.go [hunk 5]
```diff
@@ -97,3 +99,50 @@ func (is *IdentityServer) processUserProfilePicture(ctx context.Context, usr *tt
...
+			}
+		}
+		if original != "" {
+			dev.Picture.Sizes = map[uint32]string{0: original}
+		} else {
+			dev.Picture.Sizes = nil
+		}
+	}
+
+	// Embedded (uploaded) picture. Make square.
```
Comment (htdvisser) [inline pkg/identityserver/picture.go]: @mjamescompton are you aware of the fact that these images will be square?

Comment (mjamescompton) [reply pkg/identityserver/picture.go]: Should not be a problem, let's only change it if Wilhelmsen ask about it.

File: pkg/ttnpb/field_mask_validation.go [hunk 6]
```diff
@@ -36,6 +36,7 @@ var isEndDeviceReadFieldPaths = []string{
 	"version_ids.firmware_version",
 	"version_ids.hardware_version",
 	"version_ids.model_id",
+	"picture",
```
Comment (rvolosatovs) [inline pkg/ttnpb/field_mask_validation.go]: If this is here, then all subfields have to be present as well.

File: pkg/ttnpb/field_mask_validation.go [hunk 7]
```diff
@@ -55,6 +56,7 @@ var isEndDeviceWriteFieldPaths = []string{
...
```
Comment (rvolosatovs) [inline pkg/ttnpb/field_mask_validation.go]: IDEM

File: pkg/ttnpb/field_mask_validation.go (hunk 6 above)
Comment (htdvisser) [reply pkg/ttnpb/field_mask_validation.go]: I think this is actually fine the way it is. I can't think of any reason why a user would only want to get the MIME type of the picture.

File: pkg/ttnpb/field_mask_validation.go (hunk 7 above)
Comment (htdvisser) [reply pkg/ttnpb/field_mask_validation.go]: Yeah this is really fine, we only allow updating the entire picture object.

**Pull request merged**